the-collab-lab/tcl-15-smart-shopping-list
# Issue #5: Users can join existing shopping lists.

## Type of Changes

|      | Type                       |
| --- | -------------------------- |
|  ✓    | :sparkles: New feature     |
|  ✓    |  🔨  Refactoring  |

## Acceptance Criteria
## Related Issue
Closes #5

## Description
This PR does three things:

1- **It adds four new components**:
* `Header`: Just a simple header to the app, does nothing for now.
* `JoinAShoppingList`: A component that contains a form that allows users to enter shared tokens. A component that's responsible for making sure that only users with existing tokens get access to the shopping lists associated with these tokens.
* `MainIterface`: A component works like a homepage for unauthorized users. It contains a button that allows new users to generate tokens so that they can use the app. It also contains the `JoinAShoppingList` component. 
* `Form`: A component used as an attempt to reduce repetitive code.

2- **It adds a little bit of style to the app.**

3- **It removes**:
* The homeNavLink.
* The list route. 

## Demo
![a demo for the app](https://i.ibb.co/20rGwsn/demo.gif)

## Testing Steps / QA Criteria
* From your terminal, pull down this branch with `git pull origin yk-hn-issue-5` and check that branch out with `git checkout yk-hn-issue-5` 

* Run `yarn start` to run the [local server](http://localhost:3000/).

* Try to join a shopping list with a fake token and then you should see this error msg: *Shopping list does not exist*.
* Try out this token: **dahlia term thug**, which is a token that has a shopping list in the database. By submitting that token you should have access to that shopping list.
If you still see that error msg, that means that you're out of luck and that someone must've cleared the database so that token no longer exists. If that happens, just let us know and we'll give you another one. We literally have a list of like 7000 words just for these token.
